### DISCUSSION TITLE
What is the right way to use context with react

### DISCUSSION THREAD

**Discussion by eskimoblood** [statelyai/xstate#3773] [Q&A] [answered]
We have a discussion in our team on how to interpret the following sentence from the xstate react docs

> Our recommended approach for managing global state with XState and React is to use [React Context](https://reactjs.org/docs/context.html)

Some interpret it as shown in the example, to share the machine using the react context. Other see this as a hint to not use the xstate context at all. Which seems a bit contradicting, as we would loos all the nice context update functionality build into xstate. 

Could you please clarify on this, thanks.

**Comment by Andarist** [comment, marked as answer]
It probably should be rephrased to something like:
> Our recommended approach for managing global state with XState and React is to use [React Context](https://reactjs.org/docs/context.html) to pass the XState actors through the React tree.
